src/createSliceSaga.ts

```diff
@@ -131,26 +130,22 @@ export function createSliceSaga<
     string,
     ActionCreatorWithPreparedPayload<any[], any, string, never, never>
   > = {};
-  const sagas: CallEffect[] = [];
-
+  const callEffects: CallEffects = [];
   caseSagasNames.forEach((sagaName) => {
     const type = getType(name, sagaName);
     let prepareCallback: PrepareAction<any> | undefined;
     actionCreators[sagaName] = prepareCallback
       ? createAction(type, prepareCallback)
       : createAction(type);
 
-    sagas.push(
+    callEffects.push(
       call(
         sagaType === SagaType.Normal
           ? caseSagas[sagaName]
           : createSaga(sagaType, type, caseSagas[sagaName]),
       ),
     );
   });
-  function* saga(): Generator<AllEffect<CallEffect<any>>, void, unknown> {
-    yield all(sagas);
-  }
 
-  return { saga, name, actions: actionCreators as any, sagaType };
+  return { callEffects, name, actions: actionCreators as any, sagaType };
```

Comment:
It should be a "saga", not a "saga" array. 
So 
```ts
function* saga(): Generator<AllEffect<CallEffect<any>>, void, unknown> {
    yield all(sagas);
  }
``` 
is needed.

Reply:
callEffects is a saga call result array, not a saga array. I did this because I want【reduce saga nesting level, making rootSaga execution data more flat】but I don't know if is needed or correct, so I will follow you advice.

Reply:
Take a look at my example.

```ts
const sliceSaga1 = createSliceSaga(...);
const sliceSaga2 = createSliceSaga(...);
...
const sliceSagaN = createSliceSaga(...);

const rootSaga = createSagas([sliceSaga1.saga, sliceSaga2.saga, ..., sliceSagaN.saga]);
```

Reply:
I understand your intention.

It can be like this in my initial writing：
const sliceSaga1 = createSliceSaga(...);
const sliceSaga2 = createSliceSaga(...);
...
const sliceSagaN = createSliceSaga(...);
const rootSaga = createRootSaga([sliceSaga1.callEffects, sliceSaga2.callEffects, ..., sliceSagaN.callEffects]);

the different is sliceSagaX.saga is a Generator function, in sliceSagaX.saga you called call function,
sliceSagaX.callEffects is an array of Generator function call result.